trieste/acquisition/rule.py

```diff
@@ -267,8 +267,15 @@ def acquire(
         if summary_writer:
             with summary_writer.as_default(step=step_number):
                 batched_points = tf.expand_dims(points, axis=0)
-                value = self._acquisition_function(batched_points)[0][0]
-                logging.scalar("EGO.acquisition_function/maximum_found" + "[0]" * greedy, value)
+                values = self._acquisition_function(batched_points)[0]
+                if len(values) == 1:
+                    logging.scalar(
+                        "EGO.acquisition_function/maximum_found" + "[0]" * greedy, values[0]
+                    )
+                else:  # vectorized acquisition function
+                    logging.histogram(
+                        "EGO.acquisition_function/maximum_found" + "[0]" * greedy, values
+                    )
 
         if isinstance(self._builder, GreedyAcquisitionFunctionBuilder):
```

Comment:
BTW you could replace by if(greedy)

Reply:
IIRC that makes mypy sad as it can no longer figure out that `update_acquisition_function` just below accepts a `pending_points` argument.